prepare_database.go

```diff
@@ -45,7 +45,8 @@ func defaultInitDatabase(binaryExtractLocation, runtimePath, pgDataDir, username
 	postgresInitDBProcess.Stdout = logger
 
 	if err = postgresInitDBProcess.Run(); err != nil {
-		return fmt.Errorf("unable to init database using '%s': %w", postgresInitDBProcess.String(), err)
+		logContent, _ := readLogsOrTimeout(logger) // we want to preserve the original error
```

Comment:
We'll need to handle this error and bubble it.

Reply:
In theory I'm not sure I agree on the bubbling, since we are already returning a custom error from `fmt.Errorf` and I _really_ don't want the error from postgres to be overwritten by the failure to read the logs (which very well might not exist, and isn't critical if they don't). Therefore, I've attempted to meet you in the middle by adding the log reading error to the end of the log content instead of bubbling. Does that suffice?

Reply:
@fergusstrange how you feeling on this?

Reply:
Yeap I think that's a great solution.